src/InvestmentManager.sol

```diff
@@ -573,12 +576,14 @@ contract InvestmentManager is Auth {
         liquidityPool =
             liquidityPoolFactory.newLiquidityPool(poolId, trancheId, _currency, tranche.token, address(this));
 
-        EscrowLike(escrow).approve(tranche.token, liquidityPool, type(uint256).max);
         tranche.liquidityPools[_currency] = liquidityPool;
         wards[liquidityPool] = 1;
 
         // enable LP to take the liquidity pool tokens out of escrow in case if investments
         AuthLike(tranche.token).rely(liquidityPool); // add liquidityPool as ward on tranche Token
+        ERC2771Like(tranche.token).addTrustedForwarder(liquidityPool);
+        EscrowLike(escrow).approve(liquidityPool, address(this), type(uint256).max); // approve investment manager on tranche token for coordinating transfers
+        EscrowLike(escrow).approve(liquidityPool, liquidityPool, type(uint256).max); // approve liquidityPool on tranche token to be able to burn
```

Comment:
I need to better understand why this is the way it is, but will dive into that later, just leaving a note here as a reminder